cli/fs_util.rs

```diff
@@ -2,12 +2,28 @@
 
 use deno_core::error::AnyError;
 pub use deno_core::normalize_path;
+use deno_crypto::rand;
 use std::env::current_dir;
 use std::fs::OpenOptions;
 use std::io::{Error, Write};
 use std::path::{Path, PathBuf};
 use walkdir::WalkDir;
 
+pub fn atomic_write_file<T: AsRef<[u8]>>(
+  filename: &Path,
+  data: T,
+  mode: u32,
+) -> std::io::Result<()> {
+  let rand: String = (0..4)
+    .map(|_| format!("{:02x}", rand::random::<u8>()))
+    .collect();
+  let extension = format!("{}.tmp", rand);
+  let tmp_file = filename.with_extension(extension);
```

Comment:
Use `tempfile::tempfile()` https://docs.rs/tempfile/3.1.0/tempfile/fn.tempfile.html

Reply:
Tempfile could be on a different disk. This way guarantees that we don't move between disks.

Reply:
What about `tempfile_in()` https://docs.rs/tempfile/3.1.0/tempfile/fn.tempfile_in.html

Reply:
https://docs.rs/tempfile/3.1.0/tempfile/fn.tempfile_in.html

Reply:
Why though? The docs say these temp files are auto cleaned by the OS when the last handle is dropped. We don't want that though. Deno dir files should be permanent.

Reply:
Right - makes sense